jina/peapods/zmq/__init__.py

```diff
@@ -208,16 +208,17 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
-    def close(self, *args, **kwargs):
+    def close(self, force_close=False, *args, **kwargs):
         """Close all sockets and shutdown the ZMQ context associated to this `Zmqlet`.
 
         .. note::
             This method is idempotent.
 
+        :param force_close: forces a close, even if `is_closed` is already set to True
         :param args: Extra positional arguments
         :param kwargs: Extra key-value arguments
         """
-        if not self.is_closed:
+        if not self.is_closed or force_close:
```

Comment:
I guess is just for investiagtion. But if it ends up being the solution, it would be a smell. Then it would mean that the `semantics` of `is_closed` would be false

Reply:
No it is actually needed. The child class checks `self.is_closed` and sets it, if it was not closed beforehand. But the parent class should still be shut down.

Reply:
then the naming or semantics need to change.

Reply:
Not sure it's actually needed, after we realized the change was in the flags. Will get back to this when I have some time

Reply:
Let's put this as draft